### DISCUSSION TITLE
[ZK Stack] - <block_reverter help!>

### DISCUSSION THREAD

**Discussion by qiwenzhou** [zkSync-Community-Hub/zksync-developers#552] [ZK Stack]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [X] zkEVM (era-zk_evm)
- [X] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.74.0

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

I am wondering  how to use block_reverter tool to revert blocks since I ran rollback-db and encountered some error like inner state assert error. 
Literally I am asking if want to rollback to a certain block unexecuted, the exact steps or commands that need to go through.

### Expected Behavior

how to use block_reverter?

### Current Behavior

see above

### Additional Details

see above

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by qiwenzhou** [comment]
Is there any way to rollback prover status when some proof jobs fails?
<img width="720" alt="image" src="https://github.com/zkSync-Community-Hub/zksync-developers/assets/31309283/16d895c6-ad23-4a4b-9b8b-d9faab0c25dc">

**Reply by qiwenzhou** [reply]
It seems that block_reverter does not revert prover db status. When i want to run block_reverter, What is the proper way to deal with prover database?

**Reply by bxpana** [reply]
Will check on this for you.

**Comment by shahar4** [comment]
Hey @qiwenzhou, Thanks for reaching out with this question! You are correct that the block reverter won't restart the prover db. If you have failed jobs they should retry automatically. Notice that you have all the prover components running. This indicates you may not have the compressor running. All of [these steps](https://github.com/matter-labs/zksync-era/tree/main/prover/prover_fri) should be followed to run the prover.

**Reply by qiwenzhou** [reply]
Hello there. Thanks for reply! I think I have the compressor running ,or maybe another mistake by me.
Anyway, I sent the L1 tx and rollbacked the db. After that, I drop all prover db by running "zk db reset -p". For now I restart the server and seems fine so far. Is there any problem for the operations I have done?
Using sepolia, Now the tx cost is crazy hhh:
<img width="962" alt="image" src="https://github.com/zkSync-Community-Hub/zksync-developers/assets/31309283/9d8666ab-2e05-411c-9f4f-cd0a126f4b67">
doesn't look right.

**Reply by shahar4** [reply]
@qiwenzhou good job over there this sounds perfect and I'm happy to hear it's working!
